### PR TITLE
add gamespace component

### PR DESCRIPTION
**Pull request by rccsousa**
## Why
A game space is required for users to interact and check if their guesses are right / wrong / in the right direction.
## How
By adding a new `getPlayerMethods` to the `Wordle.sol` smart contract, which creates and snapshots the guess hitmap to show the player their guess hits/misses. Implements this method on the frontend by calling it and rendering the guess a cell, similar to Wordle.
## Demo

[Video](https://github.com/user-attachments/assets/4f2eefcf-4e23-4325-845c-a089e183f609)

### CODE REVIEW

File: frontend/src/abis/WordleABI.json
Comment (rccsousa) [inline frontend/src/abis/WordleABI.json:1]: updated via `forge inspect` to introduce the new method

File: frontend/src/hooks/WordleHooks.ts
Comment (rccsousa) [inline frontend/src/hooks/WordleHooks.ts:1]: created new hook to use the `getPlayerGuesses` method

File: src/Wordle.sol
Comment (rccsousa) [inline src/Wordle.sol:1]: introduces the new Guesses variable and respetive update mechanisms

**Review by DavideSilva — APPROVED** [review]
🚀

**Pull request merged**